cla_common/laalaa.py

```diff
@@ -0,0 +1,38 @@
+import requests
+
+
+class LaaLaaError(Exception):
+    pass
+
+
+class LaalaaProviderCategoriesApiClient(object):
+    instance = None
+    def __init__(self, laalaa_base_url, category_translator):
+        self.categories = {}
+        self.category_translator = category_translator
+        self.laalaa_base_url = laalaa_base_url
+
+    def get_categories(self):
+        return self.categories or self._load_categories()
+
+    def _load_categories(self):
+        categories = self._fetch_categories()
+        # Translate categories and force keys to lower case
+        for category in categories:
+            code = category["code"].lower()
+            self.categories[code] = self.category_translator(category["name"])
```

Comment:
🤔 Neither of the clients (https://github.com/ministryofjustice/cla_public/pull/872, https://github.com/ministryofjustice/fala/pull/46) seem to be using this translation functionality.

If I assume this can be removed, the class here becomes a wrapper around a `request.get.map_from_json`.

If this is a "to be used" plug to plug-in Welsh translations, what do you think about making that first-class citizen in the source API?

Since the clients have to manage the endpoint configuration and already use `requests`, would be it simpler to make the call in those directly?

Reply:
>Neither of the clients (ministryofjustice/cla_public#872, ministryofjustice/fala#46) seems to be using this translation functionality.

They both pass the `_` for the category_translator parameter.
`_` is an alias for `flask.ext.babel.lazy_gettext` or `django.utils.translation.gettext_lazy` depending on the application you're looking at.

I like the idea of moving the translation to the source api and outputting the translations from there.

>Since the clients have to manage the endpoint configuration and already use requests, would be it simpler to make the call in those directly?

I think pip doesn't store/pull down multiple copies of a dependency so I can't see the impact of using a library in multiple places. Also the currently the class caches the results of api get result. Getting rid of the shared code would mean I have to do in both the other applications.

Reply:
> `_` is an alias for `flask.ext.babel.lazy_gettext` or `django.utils.translation.gettext_lazy`

TIL 🙇 

> I think pip doesn't store/pull down multiple copies of a dependency so I can't see the impact of using a library in multiple places.

I'm not sure I understand this, sorry. I based on the comment on the extensive changes in FALA (and locking pip, etc.), but since there's a translation in both applications it's no longer important.

🤔 On that note, if there's a new category, two apps have to be updated with translations which can easily be missed as there's no warning if we can't find one.

> Also the currently the class caches the results of api get result. Getting rid of the shared code would mean I have to do in both the other applications.

It seems to me (I may misunderstand) that both applications use this code once during startup to hydrate a constant of available categories. Caching doesn't seem to be needed for our use case.

Reply:
>since there's a translation in both applications it's no longer important.

Actually I had a look at this and there is translation of categories in cla_public however fala uses on page google translate and doesn't actually have any server side translations.

>It seems to me (I may misunderstand) that both applications use this code once during startup to hydrate a constant of available categories.

You're right and that was my first approach but the flask app (cla_public) was really picky about access config parameters before application started so I couldn't place it simply in a constant at top of the app(errored with `working outside of application context`).